include/aws/http/private/hpack.h

```diff
@@ -131,7 +131,7 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
  * two header blocks. The dynamic table resize and the dynamic table size update entry will be handled properly when we
  * encode the next header block  */
 AWS_HTTP_API
-void aws_hpack_set_max_table_size(struct aws_hpack_context *context, size_t new_max_size);
+void aws_hpack_set_max_table_size(struct aws_hpack_context *context, size_t setting_max_size, bool update_table_size);
```

Comment:
blugh, don't do this, but I've always felt aws_hpack_context should be split into 2 classes aws_hpack_encoder and aws_hpack_decoder, just never spent to time to finally do it. There's nearly no overlap between the encoding/decoding parts, except for maybe this one function where the fact that it's shared makes it confusing and hard to use

maybe we should tackle this in a followup PR before we make more changes to hpack